### PR TITLE
Remove unused sort for segmenta meta list

### PR DESCRIPTION
**Pull request by k-yomo**
https://github.com/quickwit-inc/tantivy/blob/02174d26af50af2cf8b9f7592c35e90aae14de70/src/indexer/segment_updater.rs#L415-L423

The `segment_metas` method is only used in the above method, but it's re-sorted by doc count in the method as following.  So the original sort by segment id can be removed to make it little more efficient.
https://github.com/quickwit-inc/tantivy/blob/02174d26af50af2cf8b9f7592c35e90aae14de70/src/indexer/segment_updater.rs#L438

### CODE REVIEW

File: src/postings/mod.rs
Comment (k-yomo) [inline src/postings/mod.rs:193]: Since now the segment orders are only dependent on the document count, this test will be flaky.
Fixed to separate tests not to dependent on the segment id order.

File: src/indexer/segment_register.rs
Comment (fulmicoton) [inline src/indexer/segment_register.rs]:
```suggestion
        self
            .segment_states
            .values()
            .map(|segment_entry| segment_entry.meta().clone())
            .collect()
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
        let mut index_writer = index.writer_for_tests()?;
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
        let reader = index.reader()?;
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
        let mut index_writer = index.writer_for_tests()?;
```

File: src/postings/mod.rs
Comment (fulmicoton) [inline src/postings/mod.rs]:
```suggestion
        let reader = index.reader()?;
```

**Comment by fulmicoton** [comment]
LGTM Thank you @k-yomo. The explanation were very helpful too. I am waiting for the CI and will merge after.

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**
